graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/relay/PaginationHelper.java

```diff
@@ -154,10 +161,28 @@ private static <T> List<T> collectItems(Iterator<T> it, Arguments arguments, Cur
             if (arguments.before != null && cursorSupport.getCursor(value).equals(arguments.before)) {
                 break;
             }
+
+            //Adding a limit of 1000 to avoid OOM
+            if(count.intValue() == 500) {
+                logger.warn("The current paginated query is returning more than 500 items. This may cause a memory leak.");
+            } else if(count.intValue() == nodeLimit) {
+                logger.warn("The current paginated query is returning more than {} items. Stopping the query.", nodeLimit);
+                break;
```

Comment:
In case it happen, could be interesting to have debug level informations:
- Stack to have the entry point that triggered the limit reach

Would be great to have also some details about the GQL query itself for debugging purpose, but at that place it's gonna be complex to have. if we crashed an Exception it would have been easy, but we decided not to crash the execution ...

Reply:
+1